layers/+lang/typescript/packages.el

```diff
@@ -159,13 +115,14 @@
           (add-hook 'typescript-mode-hook 'spacemacs/typescript-fmt-before-save-hook))
         (spacemacs/set-leader-keys-for-major-mode 'typescript-mode
           "="  'spacemacs/typescript-format
-          "sp" 'spacemacs/typescript-open-region-in-playground)
-        (spacemacs/set-leader-keys-for-major-mode 'typescript-tsx-mode
-          "="  'spacemacs/typescript-format
-          "sp" 'spacemacs/typescript-open-region-in-playground)))))
+          "p" 'spacemacs/typescript-open-region-in-playground)))))
 
 (defun typescript/pre-init-import-js ()
   (if (eq javascript-import-tool 'import-js)
       (progn
         (add-to-list 'spacemacs--import-js-modes (cons 'typescript-mode 'typescript-mode-hook))
         (add-to-list 'spacemacs--import-js-modes (cons 'typescript-tsx-mode 'typescript-tsx-mode-hook)))))
+
+(defun typescript/post-init-tide ()
+  (add-to-list 'tide-managed-modes 'typescript-mode)
```

Comment:
`This tide` layer's implementation is almost the same as `tern-layer`. I don't like the current `lsp` layer's implementation, it's too complicated. I can add a check for `javascript-backend/typescript-backend` in this `post-init` function but I don't want to because that would defeat the ability to switch backend on the fly by setting dir local variables or `M-x: (setq javascrpipt-backend 'lsp/tide/tern)` without the need to change `.spacemacs` and restart. 

For side effects: changes made for the modes (js2-mode, typescript-modde...) this way are only applied if `tide-mode` is loaded. Think of its effects as local buffer wise. So let's say your typescript has tide as backend, and javascript has lsp as backend, that doesn't mean both `tide-mode` and `lsp-mode` are both loaded on js and ts buffers, in fact they are local buffer scope. 

Of course this is my belief combined with trial error testing so feel free to dispute :)